app/bin/tools/clear_package_properties.dart

```diff
@@ -27,68 +35,91 @@ Future main(List<String> args) async {
     return;
   }
 
+  _isDryRun = argv['dry-run'] as bool;
+  _isVerbose = argv['verbose'] as bool;
   final concurrency = int.parse(argv['concurrency'] as String);
   final package = argv['package'] as String;
   await withProdServices(() async {
     if (package != null) {
-      await _processPackage(package);
+      final p = (await dbService.lookup<Package>(
+              [dbService.emptyKey.append(Package, id: package)]))
+          .single;
+      await _processPackage(p);
     } else {
       final pool = Pool(concurrency);
       final futures = <Future>[];
 
       await for (Package p in dbService.query<Package>().run()) {
-        final f = pool.withResource(() => _processPackage(p.name,
-            packageEntityNeedsUpdate: p.additionalProperties.isNotEmpty));
+        final f = pool.withResource(() => _processPackage(p));
         futures.add(f);
       }
 
       await Future.wait(futures);
       await pool.close();
     }
   });
+
+  print(JsonEncoder.withIndent('  ').convert(_removedCount));
 }
 
 Future _processWithQuery<T extends ExpandoModel>(Query<T> query) async {
   await for (T m in query.run()) {
     if (m.additionalProperties.isNotEmpty) {
-      await dbService.withTransaction((tx) async {
-        final entry = (await tx.lookup<T>([m.key])).single;
-        entry.additionalProperties.clear();
-        tx.queueMutations(inserts: [entry]);
-        await tx.commit();
-      });
+      await _clearAdditionalProperties(m);
     }
   }
 }
 
-Future _processPackage(
-  String package, {
-  bool packageEntityNeedsUpdate = true,
-}) async {
-  print('Processing package: $package');
-  final pkgKey = dbService.emptyKey.append(Package, id: package);
+Future _processPackage(Package package) async {
+  print('Processing package: ${package.name}');
 
-  final versionQuery = dbService.query<PackageVersion>(ancestorKey: pkgKey);
+  final versionQuery =
+      dbService.query<PackageVersion>(ancestorKey: package.key);
```

Comment:
This won't work, as you can only do one transaction per second on keys with the same parent... I would assume this causes problems at some point...

Given the size of these, I'm also not sure it's wise to do all of these in one transaction.. but that might be an option.
We would likely need to merge such a transaction with `_clearAdditionalProperties(package)`.

Reply:
I've tried it on staging, and it works. It does throttle the operations one second at a time, but it does not fail. We had similar processing previously in other scripts, and it was the same behaviour.

Reply:
Okay, then let's give it a shot :)